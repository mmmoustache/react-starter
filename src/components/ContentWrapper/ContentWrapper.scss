@import '../../styles/settings.scss';
@import '../../styles/tools.scss';
/* stylelint-disable no-descending-specificity */

.root {
  padding: 40px 0;

  & > *:not(:last-child):not(:first-child) {
    margin: 20px 0;
  }

  & > *:first-child {
    margin: 0 0 20px;
  }

  & > *:last-child {
    margin: 20px 0 0;
  }
}
