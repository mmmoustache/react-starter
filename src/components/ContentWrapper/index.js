import React from 'react';
import PropTypes from 'prop-types';
import s from './ContentWrapper.scss';

const ContentWrapper = ({ children }) => <div className={s.root}>{children}</div>;

ContentWrapper.propTypes = {
  children: PropTypes.node.isRequired,
};

export default ContentWrapper;
