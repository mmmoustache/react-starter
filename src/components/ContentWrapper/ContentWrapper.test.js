import React from 'react';
import renderer from 'react-test-renderer';
import ContentWrapper from '../ContentWrapper';

describe('Component: ContentWrapper', () => {

  it('Should render successfully', () => {
    const component = renderer.create(
      <ContentWrapper>
        <p>Test</p>
        <p>Test</p>
      </ContentWrapper>
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

});