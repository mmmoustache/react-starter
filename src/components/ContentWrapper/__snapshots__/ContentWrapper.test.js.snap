// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: ContentWrapper Should render successfully 1`] = `
<div
  className={undefined}
>
  <p>
    Test
  </p>
  <p>
    Test
  </p>
</div>
`;
